client/ayon_traypublisher/addon.py

```diff
@@ -109,6 +112,13 @@ def ingestcsv(
     """
     from .csv_publish import csvpublish
 
+    # Allow user override through AYON_USERNAME when
+    # current connection is made through a service user.
+    username = os.environ.get("AYON_USERNAME")
+    con = ayon_api.get_server_api_connection()
+    if username and con.is_service_user():
+        con.set_default_service_username(username)
```

Comment:
Should we skip the API connection if the username is not passed? Just so there's potentially one less issue we may face here and it only influences the case where it is set?

```suggestion
    if username:
        con = ayon_api.get_server_api_connection()
        if con.is_service_user():
            con.set_default_service_username(username)
```

Reply:
Actually that should not be needed as python is already optimized so that second condition will not be evaluated if first one is `False`. (This is called `short-circuit` more information [there](https://docs.python.org/3/library/stdtypes.html#boolean-operations-and-or-not) is you're curious)